nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -1227,105 +1210,63 @@ define(
-                        if (!confirmed) {
-                            return;
-                        }
-
-                        // Remove all of the execution state when we reset the app.
-                        resetAppAndEdit();
+                DialogMessages.showDialog({ action: 'appReset' }).then((confirmed) => {
+                    if (confirmed) {
+                        resetToEditMode({ mode: 'editing', params: 'incomplete' });
                     }
-                );
+                });
             }
 
             /*
              * Cancelling a job is the same as deleting it, and the effect of cancelling the job is the same as re-running it.
              *
              */
             function doCancel() {
-                const confirmationMessage = div([
-                    p([
-                        'Canceling the job will halt the job processing.',
-                        'Any output objects already created will remain in your narrative and can be removed from the Data panel.',
-                    ]),
-                    p('Continue to Cancel the running job?'),
-                ]);
-                ui.showConfirmDialog({ title: 'Cancel Job?', body: confirmationMessage }).then(
-                    (confirmed) => {
-                        if (!confirmed) {
-                            return;
-                        }
-
-                        const jobState = model.getItem('exec.jobState');
-                        if (jobState) {
-                            cancelJob(jobState.job_id);
+                DialogMessages.showDialog({ action: 'cancelApp' }).then((confirmed) => {
+                    if (!confirmed) {
+                        return;
+                    }
 
-                            fsm.newState({ mode: 'canceling' });
-                            // the job will be deleted form the notebook when the job cancellation
-                            // event is received.
-                        } else {
-                            // Hmm this is a rather odd case, but it has been seen in the wild.
-                            // E.g. it could (logically) occur during launch phase (although the cancel button should not be available.)
-                            // In erroneous conditions it could occur if a job failed or was
-                            // cancelled but the state machine got confused.
-                            model.deleteItem('exec');
-                            fsm.newState({ mode: 'editing', params: 'complete', code: 'built' });
-                        }
-                        renderUI();
+                    const jobState = model.getItem('exec.jobState');
+                    if (jobState) {
+                        // the job status listener will continue to listen for updates
+                        // so the job status is tracked that way
+                        cancelJob(jobState.job_id);
+                        fsm.newState({ mode: 'canceling' });
```

Comment:
This FSM transition doesn't work as one might think/hope it does, but it works, so 🤷